yarn.lock

```diff
@@ -1939,11 +1934,16 @@
   resolved "https://registry.verdaccio.org/@verdaccio%2ftypes/-/types-8.5.2.tgz#4e371aae10e8550b4a19b57f6ba1f1f185147669"
   integrity sha512-x/sacqVndl1dXVKPd7pomea6gs9BKC+i83LDn6MEAO+tuhqWRsKC3UwztLA1YSKjNF33//7JCYMDQk6uaJ9ipw==
 
-"@verdaccio/ui-theme@0.3.9", "@verdaccio/ui-theme@^0.3.9":
+"@verdaccio/ui-theme@0.3.9":
   version "0.3.9"
```

Comment:
Hm, I currently do not understand why we install verdaccio itself as devDependency: https://github.com/verdaccio/verdaccio/blob/update-ui/package.json#L96

```
yarn why @verdaccio/ui-theme
yarn why v1.21.1
[1/4] Why do we have the module "@verdaccio/ui-theme"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "@verdaccio/ui-theme@0.3.9"
info Has been hoisted to "@verdaccio/ui-theme"
info Reasons this module exists
   - Specified in "dependencies"
   - Hoisted from "verdaccio#@verdaccio#ui-theme"
info Disk size without dependencies: "1.49MB"
info Disk size with unique dependencies: "1.49MB"
info Disk size with transitive dependencies: "1.49MB"
info Number of shared dependencies: 0
Done in 0.64s.
```

Reply:
We run tests on CLI https://github.com/verdaccio/verdaccio/pull/1624

Reply:
Couldn't we just use the (same) entry file directly? Or use `npm link` / `yarn link`?

It feels like this makes things more complicated than they should be.

Reply:
Because in this case the installed CLI might differ from the CLI / source in the current PR / commit.

At least I never needed to install a package itself to run e2e tests =)

Reply:
Using `link` would not be a E2E then. I want a real user behavior.  So, what I do in a nuthsell
- Install current working version in a registry (verdaccio@latest)
- Perform a real install from the registry into a temp folder
- I run some test over that installation I did previously

You cannot achieve that with links without runs in annoying issues, Angular, Babel, create-react-app had that, and now they use verdaccio instead.

Anyway, regarding `yarn why @verdaccio/ui-theme` I'm working on https://github.com/verdaccio/verdaccio/pull/1546 and soon we won't have the issue of mix dependencies anymore, so each package will be isolated each other.

Reply:
Ah ok, thanks for the clarification.

> * Install current working version in a registry (verdaccio@latest)

Mh, not sure why this then:

https://github.com/verdaccio/verdaccio/blob/master/package.json#L96

Reply:
Because I need a registry to publish the current version (https://github.com/verdaccio/verdaccio/pull/1624/files#diff-3828b08be4647e8c5e6282dc38a2f68eR23) , if you wonder why, is due I need to publish somewhere, I'd use the current build .. but .. then would not be reliable if the master has a bug.

Reply:
Ok, understood =)